### PR TITLE
Add e2e and lit tests for ukernel+dcg mix

### PR DESCRIPTION
**Pull request by newling**
This PR adds a test where we use a microkernel for a matmul, and directly generated code for a subsequent elementwise operation -- ie we mix ukernel with direct code generation.

### CODE REVIEW

**Comment by erwei-xilinx** [comment]
This fix should get the example to work: https://github.com/Xilinx/mlir-air/pull/656.

**Comment by newling** [comment]
Thanks @erwei-xilinx , the e2e matmul works with mixed ukernel and dcg now!

**Review by yzhang93 — COMMENTED** [review]
Could you rebase your PR and check if it passes ci?

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/lower_to_ukernel.mlir [hunk 1]
```diff
@@ -198,3 +198,57 @@ func.func @zero_fill_with_matmul(%arg0 : tensor<?x?x?x?xbf16>, %arg1 : tensor<?x
...
+      {
+      ^bb0(%in: bf16, %out: bf16):
+          %1 = arith.addf %in, %cst_1 : bf16
+          linalg.yield %1 : bf16
+      } -> tensor<?x?x?x?xbf16>
+
+  return %matmul_add : tensor<?x?x?x?xbf16>
+}
+// CHECK: func @zero_fill_matmul_elmwise
+// CHECK: iree_codegen.ukernel.generic "zero_bf16"
```
Comment (yzhang93) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/lower_to_ukernel.mlir:246]: I would include // CHECK-NOT: linalg.fill

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/lower_to_ukernel.mlir [hunk 2]
```diff
@@ -198,3 +198,57 @@ func.func @zero_fill_with_matmul(%arg0 : tensor<?x?x?x?xbf16>, %arg1 : tensor<?x
...
+// CHECK: iree_codegen.ukernel.generic "matmul_bf16_bf16"
+// CHECK: linalg.generic
+// CHECK-SAME: iterator_types = ["parallel", "parallel", "parallel", "parallel"]
+// CHECK: return
+
+
+
+
```
Comment (yzhang93) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/lower_to_ukernel.mlir:254]: Remove these empty lines.

File: build_tools/ci/cpu_comparison/run_test.sh [hunk 3]
```diff
@@ -298,18 +308,66 @@ function run_test() {
...
+  if [ $use_ukernel -ne 0 ]; then
+
+    compilation_flags="${compilation_flags} --iree-amdaie-enable-ukernels=all"
+
+    #The flag '--iree-amdaie-path-to-ukernels' currently does not work,
+    #see for example https:  // github.com/nod-ai/iree-amd-aie/issues/340
+    #Therefore we need to manually copy(or link) the mm.o file to the
+    #directory in which iree - compile is run.iree - compile is run in the
+    #output directory.Create the softlink only if it is has not already
+    #been created.
```
Comment (yzhang93) [inline build_tools/ci/cpu_comparison/run_test.sh:335]: Nit: Please add/remove spaces for this comment.
Question: Will this issue be fixed next?

**Review by yzhang93 — CHANGES_REQUESTED** [review]

File: build_tools/ci/cpu_comparison/run_test.sh (hunk 3 above)
Comment (newling) [reply build_tools/ci/cpu_comparison/run_test.sh]: I wasn't planning on looking at it next

File: build_tools/ci/cpu_comparison/run_test.sh [hunk 4]
```diff
@@ -298,18 +308,67 @@ function run_test() {
...
+
+  # TODO(newling) The following logic is copied from run_matmul_test.sh :
+  # factorize out common code in these 2 test scripts to increase
+  # maintainability.
+  if [ $use_ukernel -ne 0 ]; then
+
+    compilation_flags="${compilation_flags} --iree-amdaie-enable-ukernels=all"
+
+    # The flag '--iree-amdaie-path-to-ukernels' currently does not work,
+    # see for example https://github.com/nod-ai/iree-amd-aie/issues/340
```
Comment (yzhang93) [inline build_tools/ci/cpu_comparison/run_test.sh:332]: Add a full stop at the end of this line.

File: build_tools/ci/cpu_comparison/run_test.sh [hunk 5]
```diff
@@ -298,18 +308,67 @@ function run_test() {
...
+    # Therefore we need to manually copy (or link) the mm.o file to the
+    # directory in which iree - compile is run.iree - compile is run in the
```
Comment (yzhang93) [inline build_tools/ci/cpu_comparison/run_test.sh:334]: Still see some format issue. 
-> iree-comple is run. iree-compile ...

**Review by yzhang93 — APPROVED** [review]

**Pull request merged**